source/vibe/core/net.d

```diff
@@ -586,6 +586,41 @@ mixin(tracer);
 		return m_context.readBuffer.length > 0;
 	}
 
+	enum WaitForDataAsyncStatus {
```

Comment:
To reduce the amount of text to type when using this from the outside, I'd move this outside below the `TCPConnection` class.